src/main/java/edu/hm/hafner/analysis/parser/violations/ValgrindAdapter.java

```diff
@@ -0,0 +1,195 @@
+
+/**
+ * Parses Valgrind XML report files.
+ *
+ * @author Tony Ciavarella
+ */
+public class ValgrindAdapter extends AbstractViolationAdapter {
+    private static final long serialVersionUID = -6117336551972081612L;
+    private static final int NUMBERED_STACK_THRESHOLD = 2;
+    private static final int NO_LINE = -1;
+
+    @Override
+    ValgrindParser createParser() {
+        return new ValgrindParser();
+    }
+
+    @Override
+    Report convertToReport(final Set<Violation> violations) {
+        try (IssueBuilder issueBuilder = new IssueBuilder()) {
+            Report report = new Report();
+
+            for (Violation violation: violations) {
+                updateIssueBuilder(violation, issueBuilder);
+                issueBuilder.setCategory("valgrind:" + violation.getReporter());
+                issueBuilder.setDescription(generateDescriptionHtml(violation));
+                report.add(issueBuilder.buildAndClean());
+            }
+
+            return report;
+        }
+    }
+
+    private String generateDescriptionHtml(final Violation violation) {
+        final Map<String, String> specifics = violation.getSpecifics();
+        final JSONArray auxWhats = getAuxWhatsArray(specifics);
+
+        return
+                j2html.tags.DomContentJoiner.join(
+                        "",
+                        false,
+                        generateGeneralTableHtml(violation.getSource(), violation.getGroup(), specifics.get("tid"), specifics.get("threadname"), auxWhats),
+                        maybeGenerateStackTracesHtml(specifics.get("stacks"), violation.getMessage(), auxWhats),
+                        maybeGenerateSuppressionHtml(specifics.get("suppression"))
+                ).render();
+    }
+
+    private Tag generateGeneralTableHtml(final String executable, final String uniqueId, @CheckForNull final String threadId, @CheckForNull final String threadName, @CheckForNull final JSONArray auxWhats) {
+        ContainerTag generalTable =
+                table(
+                        attrs(".table.table-striped"),
+                        maybeGenerateTableRowHtml("Executable", executable),
+                        maybeGenerateTableRowHtml("Unique Id", uniqueId),
+                        maybeGenerateTableRowHtml("Thread Id", threadId),
+                        maybeGenerateTableRowHtml("Thread Name", threadName)
+                );
+
+        if (auxWhats != null && !auxWhats.isEmpty()) {
+            for (int auxwhatIndex = 0; auxwhatIndex < auxWhats.length(); ++auxwhatIndex) {
+                generalTable.with(maybeGenerateTableRowHtml("Auxiliary", auxWhats.getString(auxwhatIndex)));
+            }
+        }
+
+        return generalTable;
+    }
+
+    private Tag maybeGenerateStackTracesHtml(@CheckForNull final String stacksJson, final String message, @CheckForNull final JSONArray auxWhats) {
+        if (StringUtils.isBlank(stacksJson)) {
+            return iff(false, null);
```

Comment:
If there is no null object (empty container) simply return `null`. If required use a `SuppressWarnings` directive.
```suggestion
            return null;
```